pkg/volume/flexvolume/unmounter.go

```diff
@@ -44,13 +44,16 @@ func (f *flexVolumeUnmounter) TearDown() error {
 
 func (f *flexVolumeUnmounter) TearDownAt(dir string) error {
 
-	if pathExists, pathErr := util.PathExists(dir); pathErr != nil {
-		return fmt.Errorf("Error checking if path exists: %v", pathErr)
-	} else if !pathExists {
+	pathExists, pathErr := util.PathExists(dir)
+	if !pathExists {
 		glog.Warningf("Warning: Unmount skipped because path does not exist: %v", dir)
 		return nil
 	}
 
+	if pathErr != nil && !util.IsCorruptedMnt(pathErr) {
+		return fmt.Errorf("Error checking path: %v", pathErr)
+	}
+
```

Comment:
I have been thinking about - https://github.com/kubernetes/kubernetes/issues/67132 too and I am wondering if it is sensible to allow call to `unmount` to flex driver when `ENOTCONN` is raised.  @nkkashyap why do you think it is sensible to allow unmount call even if volume is known to be corrupted?

Reply:
@gnufied for some filesystems this may be required. The unmount call from the flexvolume can always fail if it cannot unmount.